pipelines/export_replay.py

```
@@ -0,0 +1,479 @@
+    if emitted:
+        _record_replayed_output_file(state, relative, emitted)
+
+
+def _member_identity(source_root: Path, relative: str) -> tuple[int, ...]:
+    path = source_root.joinpath(*relative.split("/"))
+    try:
+        entry = path.lstat()
+    except OSError as exc:
+        raise ExportError(
+            f"compose source {relative}: member cannot be inspected: {exc}"
+        ) from exc
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)